tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckDates.java

```diff
@@ -841,7 +849,15 @@ public static String checkIso8601(String path, String value) {
             if (lastField != null) {
                 int lastType = lastField.getType();
                 if (toOrder.indexOf(lastType) < toOrder.indexOf(type)) {
-                    return "Field out of order: " + lastField + "…" + field;
+                    return "Field "
+                            + lastField
+                            + " cannot come before field "
+                            + field
+                            + (isInterval
+                                    ? " in the " + intervalPosition + " part of the range"
+                                    : "")
+                            + ". A YMD calendar is special: bigger fields must come before smaller ones even when it feels unnatural in your language. "
+                            + " Change the text separating the fields as best you can.";
```

Comment:
```suggestion
                    return "Field "
                            + field
                            + " must come before field "
                            + lastField
                            + (isInterval
                                    ? " in the " + intervalPosition + " part of the range"
                                    : "")
                            + ". A YMD calendar is special: bigger fields must come before smaller ones even when it feels unnatural in your language. "
                            + " Change the text separating the fields as best you can.";
```

> A YMD calendar is special: bigger fields must come before smaller ones even when it feels unnatural in your language.
> Change the text separating the fields as best you can.

This is like an easter egg - trigger this error, and you get two sentences explaining what a YMD calendar is and giving guidance!

Perhaps all of the YMD errs should simply link somewhere on the site and not try to explain YMD?

Reply:
From the discussion in the meeting, it sounded like we really wanted to add more info directly in the error message, rather than have a link that people don't follow.

Another question: can we have markdown in error messages? If so, we could make many of them more informative.